plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/stock-status.block_theme.spec.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { test as base, expect } from '@woocommerce/e2e-playwright-utils';
+import path from 'path';
+
+/**
+ * Internal dependencies
+ */
+import {
+	TestingPost,
+	createPostFromTemplate,
+} from '../../utils/create-dynamic-content';
+
+const TEMPLATE_PATH = path.join( __dirname, './stock-status.handlebars' );
+
+const test = base.extend< {
+	dropdownBlockPostPage: TestingPost;
+	defaultBlockPostPage: TestingPost;
+} >( {
+	defaultBlockPostPage: async ( { requestUtils }, use ) => {
+		const testingPost = await createPostFromTemplate(
+			requestUtils,
+			{ title: 'Product Filter Stock Status Block' },
+			TEMPLATE_PATH,
+			{}
+		);
+
+		await use( testingPost );
+		await testingPost.deletePost();
+	},
+
+	dropdownBlockPostPage: async ( { requestUtils }, use ) => {
+		const testingPost = await createPostFromTemplate(
+			requestUtils,
+			{ title: 'Product Filter Stock Status Block' },
+			TEMPLATE_PATH,
+			{
+				attributes: {
+					displayStyle: 'dropdown',
+				},
+			}
+		);
+
+		await use( testingPost );
+		await testingPost.deletePost();
+	},
+} );
+
+test.describe( 'Product Filter: Stock Status Block', async () => {
+	test( 'By default it renders a checkbox list with the available stock statuses', async ( {
+		page,
+		defaultBlockPostPage,
+	} ) => {
+		await page.goto( defaultBlockPostPage.post.link );
+
+		const stockStatuses = page.locator(
+			'.wc-block-components-checkbox__label'
+		);
+
+		await expect( stockStatuses ).toHaveCount( 2 );
+		await expect( stockStatuses.nth( 0 ) ).toHaveText( 'In stock' );
+		await expect( stockStatuses.nth( 1 ) ).toHaveText( 'Out of stock' );
+	} );
+
+	test( 'Selecting a stock status filters the list of products', async ( {
+		page,
+		defaultBlockPostPage,
+	} ) => {
+		await page.goto( defaultBlockPostPage.post.link );
+
+		const outOfStockCheckbox = page.getByLabel( 'Out of stock' );
+		await outOfStockCheckbox.click();
+
+		// wait for navigation
+		await page.waitForURL( /.*filter_stock_status=outofstock.*/ );
+
+		const products = page.locator( '.wc-block-product' );
+
+		await expect( products ).toHaveCount( 1 );
+	} );
```

Comment:
I think overriding the `post` fixture per the `describe` block (instead of per suite) might be easier to follow for the template-fixture convention. For example:

```js
test.describe( 'Product Filter: Stock Status Block', async () => {
	test.describe( 'with default settings', () => {
		test.use( {
			page: async ( { page, requestUtils }, use ) => {
				const post = await createPostFromTemplate(
					requestUtils,
					{ title: 'Product Filter Stock Status Block' },
					TEMPLATE_PATH,
					{}
				);

				await page.goto( post.link );
				use( page );
			},
		} );

		test( 'renders a checkbox list with the available stock statuses', async ( {
			page,
		} ) => {
			const stockStatuses = page.locator(
				'.wc-block-components-checkbox__label'
			);

			await expect( stockStatuses ).toHaveCount( 2 );
			await expect( stockStatuses.nth( 0 ) ).toHaveText( 'In stock' );
			await expect( stockStatuses.nth( 1 ) ).toHaveText( 'Out of stock' );
		} );

		test( 'filters the list of products by selecting a stock status', async ( {
			page,
		} ) => {
			const outOfStockCheckbox = page.getByLabel( 'Out of stock' );
			await outOfStockCheckbox.click();

			// wait for navigation
			await page.waitForURL( /.*filter_stock_status=outofstock.*/ );

			const products = page.locator( '.wc-block-product' );

			await expect( products ).toHaveCount( 1 );
		} );
	} );

	test.describe( 'with displayStyle set to dropdown', () => {
		test.use( {
			page: async ( { page }, use ) => {
			// ...
```

This way, each scoped test starts at the target `page`, and the variants separation is done at a parent level, which allows for a more concise test title.

You might also notice that I didn't call `deletePost` in this example. It's because I think it's better to just call `requestUtils.deleteAllPosts()` [in the global setup](https://github.com/woocommerce/woocommerce/blob/417683095a20177f45b6137881167fee162626f1/plugins/woocommerce-blocks/tests/e2e/global-setup.ts#L237-L244), so that it resets the env before any test is run. This way we can drop that extra `deletePost` implementation and likely save time on the execution.

Reply:
@WunderBart I can't run `deleteAllPosts` in global setup without refactoring the existing setup where static files generate a bunch of posts relied on by multiple tests (see `bin/posts/*`). I think we can do that later as we migrate away from those static posts that get generated at test setup, though so thanks I think we can keep improving this e2e setup.

the scoping approach looks good to me though and I'll implement that.

Reply:
> I thought the point is to keep tests isolated from each other? if we don't delete the post when the test runs its technically one more change spilling over between test suites? 

Not sure I follow. With the fixture approach, each `test()` will receive a unique post the page will open with. Even if we don't delete them at all, they should not be bleeding over the other tests unless we're specifically testing, e.g., the number of posts or something in that area (which I don't think we're doing). I think we should be safe whether we remove the post or not. Does that make sense?

> There is a blocker for actually doing this though. The existing approach is that a set of posts is generated before global setup runs via wp-cli in bash scripts. (They happen as part of setting up the docker environment). So running deleteAllPosts at that point will delete the ones created during env setup. 

TIL! It does sound like a blocker. I think it's still worth pursuing though, as it's safer to cleanup in the global setup than in an after* hook, as they aren't always called. When cleaning in the after hook, all it takes to have a dangling post is to stop the test somewhere in the middle!

> As for the scoping approach, it looks good to me! I agree its more intuitive to see the context of what's under test. I'll implement that! 🙇

Nice, happy I could help! 🙌

Reply:
As a potential follow-up, it would be great to implement the `createPostFromTemplate` as an extension of core's `RequestUtils`. This way we wouldn't need to import it explicitly and the usage would be a tad simpler:

```js
test.use( {
	page: async ( { page, requestUtils }, use ) => {
		const post = await requestUtils.createPostFromTemplate(
			// no need to pass requestUtils as an argument
			{ title: 'Product Filter Stock Status Block' },
			TEMPLATE_PATH,
			{}
		);

		await page.goto( post.link );
		await use( page );
	},
} );
```

Reply:
> In the initial approach we were running multiple tests against the same post, which was likely to produce unpredictable initial state. Anyway, in the suggested approach, I think we should be safe whether we remove the post or not! Does that make sense?

Yes for sure, I just assumed you were wanting more perfect isolation per test than that, yes in future someone could be testing something meta about posts or something else, but I am not concerned about it either. I just wanted to get on same page with you in this instance 👍🏻 

>  I think it's still worth pursuing though

Yes I'm advocating to do it, but it will require refactoring many other tests. I will create a cooldown task to do that refactoring.

> As a potential follow-up, it would be great to implement the createPostFromTemplate as an extension of core's RequestUtils (possibly somew[here](https://github.com/woocommerce/woocommerce/blob/d329ced1e8aff7d5b3b4a63364930372519d1aa9/plugins/woocommerce-blocks/tests/e2e/playwright-utils/test.ts#L157-L167) here). This way we wouldn't need to import it explicitly and the usage would be a tad simpler:

Great idea. I might not get a chance to work on this again for a couple of cycles so I'll do that in this PR